runtime/common/src/parachains.rs

```diff
@@ -322,6 +429,62 @@ decl_module! {
 			Ok(())
 		}
 
+		/// Provide a proof that some validator has commited a double-vote.
+		pub fn report_double_vote(
+			origin,
+			report: DoubleVoteReport,
+		) -> DispatchResult {
+			ensure_none(origin)?;
+
+			let session_index = <session::Module<T>>::current_index();
+			let validators = <session::Module<T>>::validators();
+			let validator_set_count = validators.len() as u32;
+			let parent_hash = <system::Module<T>>::block_hash(<system::Module<T>>::block_number());
+
+			let authorities = Self::authorities();
+			let offender_idx = match authorities.iter().position(|a| *a == report.identity) {
+				Some(idx) => idx,
+				None => return Err("Not in validator set".into()),
+			};
+
+			let offender = match T::FullIdentificationOf::convert(validators[offender_idx].clone()) {
```

Comment:
At the moment, this will only work for validators who are in the current set. In the future we probably have to accept another argument to the `DoubleVoteReport` that we can use with `session::historical::Module` to extract the `FullIdentification`.